src/components/confirm/confirm.test.js

```diff
@@ -0,0 +1,386 @@
+        CypressMountWithProviders(
+          <ConfirmComponent cancelButtonDestructive={state} />
+        );
+
+        cancelButton()
+          .should("have.css", "color", color)
+          .and("have.css", "border-color", bordercolor);
+      }
+    );
+
+    it.each([
+      [true, "rgb(205, 56, 75)"],
+      [false, "rgb(0, 126, 69)"],
+    ])(
+      "should check confirm button has destructive CSS properties",
+      (state, backgroundcolor) => {
+        CypressMountWithProviders(
+          <ConfirmComponent confirmButtonDestructive={state} />
+        );
+
+        confirmButton().should("have.css", "background-color", backgroundcolor);
+      }
+    );
+
+    it("should check clicking cancel button closes dialog", () => {
+      CypressMountWithProviders(<ConfirmComponent />);
+
+      cancelButton().click();
+      dialogPreview().should("not.exist");
+    });
+
+    it("should check clicking confirm button closes dialog", () => {
+      CypressMountWithProviders(<ConfirmComponent />);
+
+      confirmButton().click();
+
+      dialogPreview().should("not.exist");
+    });
+
+    it("should check clicking confirm button does not close dialog when confirm button is disabled", () => {
+      CypressMountWithProviders(<ConfirmComponent disableConfirm />);
+
+      confirmButton().click({ force: true });
+
+      dialogPreview().should("be.visible");
+    });
+
+    it("should check clicking Esc key does not close dialog when cancel button is disabled", () => {
+      CypressMountWithProviders(<ConfirmComponent disableEscKey />);
+
+      dialogPreview().trigger("keydown", keyCode("Esc"));
+
+      dialogPreview().should("be.visible");
+    });
+
+    it("should render Confirm with aria-label prop", () => {
+      CypressMountWithProviders(<ConfirmComponent aria-label="cypress-aria" />);
+
+      dialogPreview().should("have.attr", "aria-label", "cypress-aria");
+    });
+
+    it("should check confirm button has save icon", () => {
+      CypressMountWithProviders(
+        <ConfirmComponent confirmButtonIconType="Save" />
+      );
+
+      getDataElementByValue("Save").should("have.attr", "type", "Save");
+    });
+
+    it("should check cancel button has bin icon", () => {
+      CypressMountWithProviders(
+        <ConfirmComponent cancelButtonIconType="bin" />
+      );
+
+      getDataElementByValue("bin").should("have.attr", "type", "bin");
+    });
+
+    it("should render Confirm with aria-labelledby prop", () => {
+      CypressMountWithProviders(
+        <Confirm open aria-labelledby={testPropValue} />
+      );
+
+      dialogPreview().should("have.attr", "aria-labelledby", testPropValue);
+    });
+
+    it("should render Confirm with aria-describedby prop", () => {
+      CypressMountWithProviders(
+        <Confirm open aria-describedby={testPropValue} />
+      );
+
+      dialogPreview().should("have.attr", "aria-describedby", testPropValue);
+    });
+
+    it("should focus first element", () => {
```

Comment:
I think we should also have a test for when the `focusFirstElement` is not set and one that shows that the focus loops between the elements if possible. However, as the functionality of the `FocusTrap` is being updated I'd be okay with a `TODO` being added so we can cover that functionality when https://github.com/Sage/carbon/pull/5117 is merged